src/progress/progress_set.rs

```diff
@@ -195,51 +199,72 @@ impl ProgressSet {
                 DefaultHashBuilder::default(),
             ),
             sort_buffer: RefCell::from(Vec::with_capacity(voters)),
-            configuration_capacity: (voters, learners),
             configuration: Configuration::with_capacity(voters, learners),
             next_configuration: Option::default(),
             logger: logger.new(o!()),
         }
     }
 
+    fn clear(&mut self) {
+        self.progress.clear();
+        self.configuration.voters.clear();
+        self.configuration.learners.clear();
+        self.next_configuration = None;
+    }
+
+    pub(crate) fn restore_conf_states(
+        &mut self,
+        conf_states: &[ConfStateWithIndex],
+        next_idx: u64,
+        max_inflight: usize,
+    ) {
+        if conf_states.is_empty() {
+            return;
+        }
+        let mut meta = SnapshotMetadata::default();
+        let len = conf_states.len();
+        if conf_states[len - 1].in_membership_change {
+            meta.set_conf_state(conf_states[len - 2].conf_state.clone());
```

Comment:
I worry about slices of length `1`, do you think it's possible?

Reply:
By design, if the peer is in membeship change, the `conf_states` must have 2 elements at least. Otherwise it must be a bug.